docs/model_developers/modules.rst

```diff
@@ -122,3 +122,60 @@ The format for requires in requirements.txt is the folllowing:
  * Or a repository location such as  git+https://github.com/project/python-foo The correct syntax
    to use is then: eggname@git+https://../repository#branch with branch being optional.
 
+
+Working on modules
+==================
+Since modules often have dependencies on other modules, it is common to develop against multiple
+modules (or a project and one or more modules) simultaneously. One might for example need to
+extend a dependency module to add support for some new feature. Because this use case is so common,
+this section will describe how to work on multiple modules simultaneously so that any changes are
+visible the compiler. This procedure is of course applicable for working on a single module as well.
+
+Setting up the dev environment
```

Comment:
My general comment on this section is that we should make this have less `branches`.

This is external documentation, so I think we need an explanation that allows one to copy paste some  command and end up with a dev environment, without having to know exactly what is what. 

We can then in second instance elaborate on alternatives and nuances. 

I would assume, in first instance, that we are working in the context of a project, in a venv (as is in the developer getting started docs iirc), with a mix of packaged and unpackaged modules. 

In second instance we can add nuances such as e.g. having a single libs folder for multiple projects (by updating the search path in the project file)

Reply:
All right, I'll see what I can do. To what extent should I account for v1 modules in this initial focus?

Reply:
The process should preferably be identical for both.